gen/internals.py

```diff
@@ -602,7 +668,7 @@ def _calculate_target(self, target):
             if resolvable.is_error:
                 continue
 
-            assert resolvable.is_resolved, "uhhh: {}".format(resolvable)
+            assert resolvable.is_resolved or resolvable.is_late, "uhhh: {}".format(resolvable)
```

Comment:
Should update my comment from `uhhh` to something more indicative what went wrong "Internal logic error: Variable was not resolved to a value or found to be late, but somehow we got out of the variable value calculation logic without an error."